File: rust/axum/src/modules/transaction/module.rs
use axum::{
    middleware::from_fn, routing::{get, post}, Router
};

use crate::modules::auth::middleware::auth_middleware;
use crate::modules::transaction::controller::TransactionController;

pub fn transaction_module() -> Router {
    let router =Router::new()
        .route("/", get(TransactionController::find_all)).layer(from_fn(auth_middleware))
        .route("/", post(TransactionController::create)).layer(from_fn(auth_middleware));

    Router::new().nest("/transacoes", router)
} 

File: go/fiber_orm/internal/modules/transaction/module.go
package transaction

import (
	"tcc/internal/modules/account"
	auth "tcc/internal/modules/auth/middleware"

	"github.com/gofiber/fiber/v2"
)

func TransactionModule(router *fiber.App) {
	service := TransactionService{
		repository: struct{account account.AccountRepository; transaction TransactionsRepository}{
			account: account.AccountRepository{},
			transaction: TransactionsRepository{},
		},
	}
	controller := TransactionController{
		service: service,
	}

	group := router.Group("/transacoes")
	{
		group.Get("", auth.AuthMiddleware(), controller.FindAll)
		group.Post("", auth.AuthMiddleware(), controller.Create)
	}
}